### PR TITLE
fix IMU sensor readings

### LINKED ISSUE #33721: simulator: sensors aren't filled
The IMUState is never updated, so locationd is getting 0s for the sensor data, and then paramsd learns a bad offset, preventing from merging a fix to the steering angle: https://github.com/commaai/openpilot/pull/33718

https://github.com/commaai/openpilot/blob/72e19ccfc6de0a145bf778fdc701878f969347bb/tools/sim/lib/common.py#L30-L34

### PR DESCRIPTION
**Pull request by sambhavnoobcoder**
## Description
The simulator's IMU sensors were not being populated with realistic values, causing issues with:
- locationd calibration 
- paramsd steering angle offset learning
- overall vehicle control accuracy

This led to:

- locationd receiving zero values for sensor data
- paramsd learning incorrect steering offsets

## Proposed Solution & Implementation
Implemented IMU sensor value calculation using vehicle state data from MetaDrive. The solution:

- Added IMU calculation logic in read_sensors()
- Added IMU calculation function that:

- Computes realistic accelerometer values considering:
  - Linear acceleration
  - Centripetal forces during turns
  - Gravitational acceleration (9.81 m/s²)

- Calculates gyroscope values from bearing changes
- Applies appropriate coordinate transformations
- Enforces realistic physical limits on sensor values

## Testing and Validation
-  For testing , wrote custom unit tests to ensure consistency , robustness and functionality .
- Test file can be currently found here : https://github.com/sambhavnoobcoder/openpilot/blob/Testing-Sensors-Bounty/tools/sim/tests/test_metadrive_world.py . 
**NOTE:** This file was not added in this PR because - 
a) This appeared out of scope of the bounty 
b) more importantly , it made the PR big in LOC . will add this this as a separate PR if required .  
- The implementation:

1. Successfully generates non-zero, physically realistic IMU values
2. Maintains accelerometer readings within vehicle limits (<18 m/s²)
3. Produces reasonable gyroscope values during turns
4. Enables proper locationd calibration
5. Allows paramsd to learn accurate steering offsets
6. Improves overall steering angle correction

## Sample output and Inference
Here is a small screen recording of the outputs generated:

https://github.com/user-attachments/assets/00c3318f-e4d5-495b-8bf6-34387995118b

This output was generated with the parameters : 
```
metadrive_vehicle_state(
    velocity=vec3(x=3.25, y=10.50, z=0), 
    position=(217.06, 82.06),
    bearing=72.51,
    steering_angle=7.87
)
```
The inference drawn from this output was : 
- Accelerometer shows expected acceleration patterns during turns
- Gyroscope correctly reflects vehicle rotation rate
- Calculated values enable proper steering offset learning (previous invalid 0° → now realistic ~2.7°)

## Impact
This fix:

- Enables proper locationd calibration in simulator
- Improves steering angle offset learning
- Enhances simulation accuracy

The implementation maintains clean code organization while adding essential simulator functionality, making the simulation environment more realistic and useful for testing vehicle control systems.
Related Issues

- Fixes #33721 
- Unblocks #33718

### CODE REVIEW

File: tools/sim/bridge/metadrive/metadrive_world.py [hunk 1]
```diff
@@ -9,124 +9,215 @@
...
 from openpilot.tools.sim.bridge.metadrive.metadrive_process import (metadrive_process, metadrive_simulation_state,
                                                                     metadrive_vehicle_state)
-from openpilot.tools.sim.lib.common import SimulatorState, World
+from openpilot.tools.sim.lib.common import SimulatorState, World , vec3
 from openpilot.tools.sim.lib.camerad import W, H
+from dataclasses import dataclass
+import math
 
 
-class MetaDriveWorld(World):
```
Comment (sshane) [inline tools/sim/bridge/metadrive/metadrive_world.py:16]: You've inflated the diff by changing tabs to 4 spaces, can you reformat the file?

Comment (sambhavnoobcoder) [reply tools/sim/bridge/metadrive/metadrive_world.py:16]: apologies , didn't realise tab for 2 spaces . pushed the fix in [1ff2d0c](https://github.com/commaai/openpilot/pull/33836/commits/1ff2d0ce6e23ed10d4fa195c162a74353c8f733d) commit . this makes the diffs more clearer .

**Comment by sambhavnoobcoder** [comment]
@sshane can you review this ? else this will close in 2 days .

**Comment by sambhavnoobcoder** [comment]
@sshane could you review this please ?

**Comment by sambhavnoobcoder** [comment]
any changes needed here ?

**Pull request closed without merging**